internal/controllers/secrets.go

```diff
@@ -127,6 +129,33 @@ func (r *CryostatReconciler) reconcileJMXSecret(ctx context.Context, cr *operato
 	})
 }
 
+// databaseSecretNameSuffix is the suffix to be appended to the name of a
+// Cryostat CR to name its JMX credentials Database secret
+const databaseSecretNameSuffix = "-jmx-credentials-db"
+
+// dbSecretUserKey indexes the password within the Cryostat JMX credentials Database Secret
+const databaseSecretPassKey = "CRYOSTAT_JMX_CREDENTIALS_DB_PASSWORD"
+
+func (r *CryostatReconciler) reconcileDatabaseSecret(ctx context.Context, cr *operatorv1beta1.Cryostat) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + databaseSecretNameSuffix,
+			Namespace: cr.Namespace,
+		},
+	}
+	return r.createOrUpdateSecret(ctx, secret, cr, func() error {
+		if secret.StringData == nil {
+			secret.StringData = map[string]string{}
+		}
+
+		// Password is generated, so don't regenerate it when updating
+		if secret.CreationTimestamp.IsZero() {
+			secret.StringData[databaseSecretPassKey] = r.GenPasswd(32)
```

Comment:
What is the lifecycle of this secret? Would it be generated once when the Operator is installed or is it dropped and regenerated for each Cryostat CR instance?

Reply:
Oh, this secret is generated once (i.e. its password) for each Cryostat CR and garbage collected if CR is deleted.

Reply:
Hmm it's possible that Cryostat will re-gain access to old persistent volume (i.e. maybe after re-creation). Then, the new password can't be used. So, the user has to clean up the volume first?

Reply:
Either the user has to clean up the volume or the user can have the ability to specify their own password. I guess that would work by the user creating a Secret manually and linking to that in the CR. This way the Secret contains the password long-term, outside of the Cryostat/Operator lifecycles, so it can be consistent for different Cryostat deployments and instances if the user needs that.

Reply:
Right, this is most ideal. I think we can a spec in Cryostat CR for this and remove the managed secret now. Probably, have to document the required key name.

Reply:
Perhaps we should still generate the secret if not provided. Adding a new mandatory user-supplied field to the CRD would break the upgrade process.

Reply:
Oh yes I was concerned about that. Okayyy let me add back the secret reconciler for this password.

Reply:
Hmm, this means updates to Cryostat CR must reserve the option to whether specify secret or auto-generated? Switching between them across updates creates a mismatch in password for the database.

Reply:
That likely would not be easy to do. I think it's fine to not allow switching for now. If users are upgrading from 2.1, they'll get the auto-generated password. If they're creating a fresh CR, they can provide their own password.